src/tennissimulator/menu/Menu.java

```diff
@@ -100,61 +107,55 @@ private Gender resolveGender(String gender) {
 
     private int promptUserForRating(Organization organization) {
         System.out.print("Enter a " + organization.toString() + " rating: ");
+
         return scanner.nextInt();
     }
 
     private String promptUserForName() {
         System.out.print("Enter a full player name: ");
+
         return scanner.nextLine();
     }
 
     private String promptUserForGender() {
         System.out.print("Enter a gender (man/woman): ");
+
         return scanner.nextLine();
     }
 
-    private void selectPlayersForMatchSimulation() {
-        playersDatabase.print();
-        if (playersDatabase.getPlayers().isEmpty()) {
-            return;
-        }
+    private String promptUserForId() {
+        System.out.println("\nChoose player ID: ");
 
-        Player firstPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(firstPlayer)) {
-            return;
-        }
+        return scanner.nextLine();
+    }
 
-        Player secondPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(secondPlayer)) {
+    private void selectPlayersForMatchSimulation() throws RuntimeException {
+        playersDatabase.print();
+        if (playersDatabase.getPlayers().isEmpty()) {
             return;
         }
 
-        match = new Match(firstPlayer, secondPlayer);
-        System.out.println("\n" + firstPlayer.getName() + " is playing against "
-                + secondPlayer.getName() + "\n");
-    }
-
-    private boolean isValidPlayer(Player player) {
-        if (player == null) {
-            System.out.println("Can`t find player in database. Please enter valid name.");
-            return false;
+        Optional<Player> firstPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        Optional<Player> secondPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        if (firstPlayer.isEmpty() || secondPlayer.isEmpty()) {
+            throw new RuntimeException();
+        } else {
```

Comment:
Else branch is not needed at all, because if exception is not thrown it will be executed anyway.

Reply:
noted